src/core/public/apm_system.test.ts

```diff
@@ -0,0 +1,122 @@
+      await apmSystem.setup();
+      expect(apm.addLabels).toHaveBeenCalledWith({ alpha: 'one' });
+    });
+
+    describe('http request normalization', () => {
+      let windowSpy: any;
+
+      beforeEach(() => {
+        windowSpy = jest.spyOn(global as any, 'window', 'get').mockImplementation(() => ({
+          location: {
+            protocol: 'http:',
+            hostname: 'mykibanadomain.com',
+            port: '5601',
+          },
+        }));
+      });
+
+      afterEach(() => {
+        windowSpy.mockRestore();
+      });
+
+      it('adds a filter function', async () => {
+        const apmSystem = new ApmSystem({ active: true });
+        await apmSystem.setup();
+        expect(apm.addFilter).toHaveBeenCalled();
+      });
+
+      it('removes the hostname, port, and protocol only when all match window.location', async () => {
+        const apmSystem = new ApmSystem({ active: true });
+        await apmSystem.setup();
+        const filter = apmMock.addFilter.mock.calls[0][0];
+        const filteredPayload = filter({
+          transactions: [
+            { type: 'http-request', name: 'GET http://mykibanadomain.com:5601/asdf/qwerty' },
+            { type: 'http-request', name: 'GET https://mykibanadomain.com:5601/asdf/qwerty' },
+            { type: 'http-request', name: 'GET http://mykibanadomain.com:9200/asdf/qwerty' },
+            { type: 'http-request', name: 'PATCH http://myotherdomain.com:5601/asdf/qwerty' },
+          ],
+        });
+
+        expect(filteredPayload).toEqual({
+          transactions: [
+            { type: 'http-request', name: 'GET /asdf/qwerty' },
+            { type: 'http-request', name: 'GET https://mykibanadomain.com:5601/asdf/qwerty' },
+            { type: 'http-request', name: 'GET http://mykibanadomain.com:9200/asdf/qwerty' },
+            { type: 'http-request', name: 'PATCH http://myotherdomain.com:5601/asdf/qwerty' },
+          ],
+        });
+      });
+
+      it('strips the basePath', async () => {
+        const apmSystem = new ApmSystem({ active: true }, '/alpha');
+        await apmSystem.setup();
+        const filter = apmMock.addFilter.mock.calls[0][0];
+        const filteredPayload = filter({
+          transactions: [
+            { type: 'http-request', name: 'GET http://mykibanadomain.com:5601/alpha' },
+            { type: 'http-request', name: 'GET http://mykibanadomain.com:5601/alpha/' },
+            { type: 'http-request', name: 'GET http://mykibanadomain.com:5601/alpha/beta' },
+            { type: 'http-request', name: 'GET http://mykibanadomain.com:5601/alpha/beta/' },
+          ],
+        });
+
+        expect(filteredPayload).toEqual({
+          transactions: [
+            { type: 'http-request', name: 'GET /' },
+            { type: 'http-request', name: 'GET /' },
+            { type: 'http-request', name: 'GET /beta' },
+            { type: 'http-request', name: 'GET /beta/' },
```

Comment:
@vigneshshanmugam One question I have was whether or not we have any features in the APM UI or planned that depend on the format of the `name` field on `http-request` transactions. If there's any precedent here, I'd like to avoid breaking this. Instead of stripping the proto/host/port, we could replace it with a faux one.

Reply:
As far as i know, it should not break any assumptions. @sqren WDYT?

Reply:
No, we don't rely on any particular format.